Common/SecurityIdentifier.cs

```diff
@@ -392,9 +394,10 @@ public SecurityIdentifier(string symbol, ulong properties, SecurityIdentifier un
             string market,
             decimal strike,
             OptionRight optionRight,
-            OptionStyle optionStyle)
+            OptionStyle optionStyle,
+            SecurityType optionType = SecurityType.Option)
```

Comment:
`SecurityType optionType = SecurityType.Option` is actually provided through the underlying I think, suggest we call the `SecurityType Symbol.GetOptionTypeFromUnderlying()` so this similar logic is centralized, maybe we could move the method here?